esp-hal/src/gpio/mod.rs

```diff
@@ -202,6 +202,12 @@ impl From<Level> for bool {
     }
 }
 
+/// A configuration error.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[non_exhaustive]
+pub enum ConfigError {}
```

Comment:
I'm wondering if this should be split into seperate config error types for each pin driver :thinking:.

In the future it might be weird to have to match against input specific errors when applying a config to an output driver, and visa versa?

Reply:
After a day of thinking about this, I think they should be separate errors imo, at the very least I think we should have `InputConfigError` and `OutputConfigError`, where `Output` and `OutputOpenDrain` share `OutputConfigError`.

Reply:
#3016 is related to `OutputOpenDrain`, and whether it should exist or be merged into the current `Output` driver as I understand it. I now see the discussion about the initial level also being applicable to `Output`, though that would seem to contradict the first point there.

Regardless, `Input` and `Output` are likely to have separate error conditions, so I don't think it makes sense for both to share one error type?

I looked at what esp-idf does, and whilst it has one error type, it also doesn't make a distinction between and input and output pin, that itself is handled in the config.

Reply:
Added a comment in https://github.com/esp-rs/esp-hal/issues/3016#issuecomment-2607373052 maybe we can make this completely infallible, worth exploring at least.

With that in mind, we can leave this how it is for now then.